packages/eslint-plugin/lib/config/react.ts

```diff
@@ -1,89 +1,93 @@
-                "react/no-typos": "error",
-                "react/style-prop-object": "warn",
-                "react/button-has-type": "warn",
-                "react/destructuring-assignment": [
-                    "warn",
-                    "always",
-                    { ignoreClassFields: true }
-                ],
-                "react/jsx-boolean-value": ["warn", "never"],
-                "react/default-props-match-prop-types": "warn",
-                "react/no-unused-state": "warn",
-                "react/no-array-index-key": "warn",
-                "react/no-access-state-in-setstate": "warn",
-                "react/jsx-filename-extension": ["warn", { "extensions": [".jsx", ".tsx"] }],
-                "react/jsx-curly-brace-presence": "warn",
-                "react/no-unused-prop-types": [
-                    "warn",
-                    { customValidators: [], skipShapeProps: true }
-                ],
+            // extra react rules
+            "react/forbid-foreign-prop-types": ["warn", { allowInPropTypes: true }],
+            "react/jsx-pascal-case": [
+                "warn",
+                {
+                    allowAllCaps: true,
+                    ignore: []
+                }
+            ],
+            "react/no-typos": "error",
+            "react/style-prop-object": "warn",
+            "react/button-has-type": "warn",
+            "react/destructuring-assignment": [
+                "warn",
+                "always",
+                { ignoreClassFields: true }
+            ],
+            "react/jsx-boolean-value": ["warn", "never"],
+            "react/default-props-match-prop-types": "warn",
+            "react/no-unused-state": "warn",
+            "react/no-array-index-key": "warn",
+            "react/no-access-state-in-setstate": "warn",
+            "react/jsx-filename-extension": ["warn", { "extensions": [".jsx", ".tsx"] }],
+            "react/jsx-curly-brace-presence": "warn",
+            "react/no-unused-prop-types": [
+                "warn",
+                { customValidators: [], skipShapeProps: true }
+            ],
 
-                "react/jsx-closing-bracket-location": [1, "line-aligned"],
-                "react/jsx-tag-spacing": ["warn", { beforeSelfClosing: "always" }],
-                "react/jsx-max-props-per-line": [
-                    "warn",
-                    { maximum: 1, when: "multiline" }
-                ],
-                "react/jsx-curly-spacing": ["warn", { children: true, when: "never" }]
+            "react/jsx-closing-bracket-location": [1, "line-aligned"],
+            "react/jsx-tag-spacing": ["warn", { beforeSelfClosing: "always" }],
+            "react/jsx-max-props-per-line": [
+                "warn",
+                { maximum: 1, when: "multiline" }
+            ],
+            "react/jsx-curly-spacing": ["warn", { children: true, when: "never" }]
 
-            }
         }
-    ]
-};
+    },
+    {
+        name: "Workleap/React",
```

Comment:
Is it legit to have 2 objects with the same name?

Reply:
What are the benefits of separating the react and hooks configs?

Reply:
The double name is legit, the rules will both show up as being part of the React config. There's no functional up or down side to keeping them separate. Just makes it easier to see which is which.